Sources/StreamChat/StateLayer/UserSearchState.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Represents a list of user search results.
+@available(iOS 13.0, *)
+public final class UserSearchState: ObservableObject {
+    /// The search query the results corresponds to.
+    @Published public private(set) var searchQuery: UserListQuery?
+    
+    /// An array of search results for the specified query and pagination state.
+    @Published public private(set) var users = StreamCollection<ChatUser>([])
```

Comment:
I also considered:
- renaming it to `results`
- having native Swift array type (but then it is not aligned with other `XYZState` types)
WDTY?

Reply:
I think you made good decisions, both of them are consistent with the rest of the codebase.